java/vector/src/main/java/org/apache/arrow/vector/FieldVector.java

```diff
@@ -60,6 +60,17 @@ public interface FieldVector extends ValueVector {
    */
   List<ArrowBuf> getFieldBuffers();
 
+
+  /**
+   * Get the buffers for C Data Interface, (same size as getFieldVectors() since it is their content).
+   * By default, it returns the same as getFieldBuffers().
+   *
+   * @return the buffers containing the data for this vector (ready for exporting through C Data Interface)
+   */
+  default List<ArrowBuf> getCDataBuffers() {
+    return getFieldBuffers();
+  }
```

Comment:
I thought a few approaches to achieve the goal to fix this issue. This is less intrusive one I can think of.

I've tried to modify `getFieldBuffers` directly. But it seems also used not only for C Data Interface. So that change causes some test failures.